Tribler/Test/Core/CreditMining/test_creditmining.py

```diff
@@ -135,6 +139,57 @@ def tearDown(self):
 
         super(TestBoostingManagerUtilities, self).tearDown()
 
+    def test_boosting_dependencies(self):
+        """
+        Test whether boosting manager dependencies works or not.
+
+        In all test, check dependencies always off. In production, it is on by default.
+        """
+        self.bsettings.check_dependencies = True
+        self.bsettings.initial_swarm_interval = 9000
+        self.bsettings.initial_tracker_interval = 9000
+        self.bsettings.initial_logging_interval = 9000
+        self.session.open_dbhandler = lambda _: None
+        self.session.lm.ltmgr = MockLtSession()
+
+        self.session.get_torrent_checking = lambda: True
+        self.session.get_dispersy = lambda: True
+        self.session.get_torrent_store = lambda: True
+        self.session.get_enable_torrent_search = lambda: True
+        self.session.get_enable_channel_search = lambda: True
+        self.session.get_megacache = lambda: False
+
+        self.assertRaises(AssertionError, bm.BoostingManager, self.session, self.bsettings)
+
+    @deferred(timeout=5)
+    def test_load_default(self):
+        """
+        Test load default configuration in BoostingManager
+        """
+        self.bsettings.load_config = True
+        self.bsettings.auto_start_source = False
+        self.bsettings.initial_swarm_interval = 9000
+        self.bsettings.initial_tracker_interval = 9000
+        self.bsettings.initial_logging_interval = 9000
+        self.session.open_dbhandler = lambda _: None
+        self.session.lm.ltmgr = MockLtSession()
+
+        # it will automatically load the default configuration
+        boost_man = bm.BoostingManager(self.session, self.bsettings)
+
+        def validate(d_defer):
+            self.assertEqual(sessdefaults['credit_mining']['source_interval'], boost_man.settings.source_interval)
+            self.assertEqual(sessdefaults['credit_mining']['archive_sources'],
+                             [source_to_string(src.source) for src in boost_man.boosting_sources.values() if src.archive])
+
+            d_defer.callback(None)
+            boost_man.cancel_all_pending_tasks()
+
+        d = Deferred()
+        reactor.callLater(3, validate, d)
```

Comment:
Because the loading process is not returning deferreds. In this function, I just want to make sure the configuration can be read with some amount of time.

I changed this just now in the newest commit.